sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/TelemetryPartBTests.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using OpenTelemetry.Trace;
+using Xunit;
+
+namespace Azure.Monitor.OpenTelemetry.Exporter.Demo.Tracing
+{
+    public class TelemetryPartBTests
+    {
+        private const string ActivitySourceName = "TelemetryPartBTests";
+        private const string ActivityName = "TestActivity";
+
+        static TelemetryPartBTests()
+        {
+            Activity.DefaultIdFormat = ActivityIdFormat.W3C;
+            Activity.ForceDefaultIdFormat = true;
+
+            var listener = new ActivityListener
+            {
+                ShouldListenTo = _ => true,
+                Sample = (ref ActivityCreationOptions<ActivityContext> options) => ActivitySamplingResult.AllData,
+            };
+
+            ActivitySource.AddActivityListener(listener);
+        }
+
+        [Fact]
+        public void ValidateHttpRequestData()
+        {
+            using ActivitySource activitySource = new ActivitySource(ActivitySourceName);
+            using var activity = activitySource.StartActivity(
+                ActivityName,
+                ActivityKind.Server,
+                parentContext: new ActivityContext(ActivityTraceId.CreateRandom(), ActivitySpanId.CreateRandom(), ActivityTraceFlags.Recorded),
+                startTime: DateTime.UtcNow);
+            activity.Stop();
+            var httpUrl = "https://www.foo.bar/search";
+            var httpResponsCode = "0";
+            activity.SetStatus(Status.Ok);
+            activity.SetTag(SemanticConventions.AttributeHttpUrl, httpUrl);
+            activity.SetTag(SemanticConventions.AttributeHttpStatusCode, null);
+            var requestData = TelemetryPartB.GetRequestData(activity);
+            Assert.Equal(requestData.Name, activity.DisplayName);
+            Assert.Equal(requestData.Id, activity.Context.SpanId.ToHexString());
+            Assert.Equal(requestData.Url, httpUrl);
+            Assert.Equal(requestData.ResponseCode, httpResponsCode);
```

Comment:
nit:
1. convention - put the expected result as the 1st argument `Assert.Equal(expectedValue, actualValue)`.
2. consider removing `var httpResponseCode = "0"` and directly put the literal constant inline.